components/webfield/ProgramChairWithSeniorReviewerConsole.js

```diff
@@ -455,6 +456,279 @@ const Overview = ({ pcConsoleData }) => {
+        if (!allReviewerIds?.length) return Promise.resolve()
+        const reviewerIds = allReviewerIds.flatMap((p) =>
+          p.isSeniorReviewer ? [] : p.reviewerProfileId
+        )
+        const seniorReviewerIds = allReviewerIds.flatMap((p) =>
+          p.isSeniorReviewer ? p.reviewerProfileId : []
+        )
+        const forumReviewUrl = `https://openreview.net/forum?id=${rowData.forum}&noteId=${noteId}&invitationId=${venueId}/${submissionName}${rowData.number}/-/${officialReviewName}`
+        const forumSeniorReviewUrl = `https://openreview.net/forum?id=${rowData.forum}&noteId=${noteId}&invitationId=${venueId}/${submissionName}${rowData.number}/-/${seniorOfficialReviewName}`
+        return [
+          reviewerIds?.length
+            ? api.post(
+                '/messages',
+                {
+                  groups: reviewerIds,
+                  subject,
+                  message: message.replaceAll('{{submit_review_link}}', forumReviewUrl),
+                  parentGroup: `${venueId}/${submissionName}${rowData.number}/Reviewers`,
+                  replyTo: emailReplyTo,
+                },
+                { accessToken }
+              )
+            : null,
+          seniorReviewerIds?.length
+            ? api.post(
+                '/messages',
+                {
+                  groups: seniorReviewerIds,
+                  subject,
+                  message: message.replaceAll('{{submit_review_link}}', forumSeniorReviewUrl),
+                  parentGroup: `${venueId}/${submissionName}${rowData.number}/Senior_Reviewers`,
+                  replyTo: emailReplyTo,
+                },
+                { accessToken }
+              )
+            : null,
+        ]
+      })
+      await Promise.all(sendEmailPs.flat())
+
+      $(`#${messageModalId}`).modal('hide')
+      promptMessage(`Successfully sent ${totalMessagesCount} emails`)
+    } catch (apiError) {
+      setError(apiError.message)
+    }
+    setIsSending(false)
+  }
+
+  const getRecipients = (selecteNoteIds) => {
+    if (!selecteNoteIds.length) return []
+    const selectedRows = tableRowsDisplayed.filter((row) =>
+      selecteNoteIds.includes(row.note.id)
+    )
+
+    switch (messageOption.value) {
+      case 'allReviewers':
+        return selectedRows.flatMap((row) =>
+          row.reviewers.map((reviewer) => ({ ...reviewer, isSeniorReviewer: false }))
+        )
+      case 'allSeniorReviewers':
+        return selectedRows.flatMap((row) =>
+          row.seniorReviewers.map((reviewer) => ({ ...reviewer, isSeniorReviewer: true }))
+        )
+      case 'allReviewersAndSeniorReviewers':
+        return selectedRows.flatMap((row) => [
+          ...row.reviewers.map((reviewer) => ({ ...reviewer, isSeniorReviewer: false })),
+          ...row.seniorReviewers.map((reviewer) => ({ ...reviewer, isSeniorReviewer: true })),
+        ])
+      case 'withReviews':
+        return selectedRows
+          .flatMap((row) => row.reviewers)
```

Comment:
It seems inefficient to have 2 sequential calls to flatMap, when you could just combine them into 1 function, right?

Reply:
i think it's the same?
the 2nd flatmap is used here as filter by hasReview + map to add isSeniorReviewer
so it will become nested flatmap if combined into 1 function

Reply:
Yeah a nested map wouldn't really be better